cmd/fs-v1.go

```diff
@@ -563,17 +563,19 @@ func (fs *FSObjects) defaultFsJSON(object string) fsMetaV1 {
 // getObjectInfo - wrapper for reading object metadata and constructs ObjectInfo.
 func (fs *FSObjects) getObjectInfo(ctx context.Context, bucket, object string) (oi ObjectInfo, e error) {
 	fsMeta := fsMetaV1{}
-	fi, err := fsStatDir(ctx, pathJoin(fs.fsPath, bucket, object))
-	if err != nil && err != errFileAccessDenied {
-		return oi, err
-	}
-	if fi != nil {
-		// If file found and request was with object ending with "/", consider it
-		// as directory and return object info
-		if hasSuffix(object, slashSeparator) {
-			return fsMeta.ToObjectInfo(bucket, object, fi), nil
+	if fsIsDir(ctx, pathJoin(fs.fsPath, bucket, object)) {
+		fi, err := fsStatDir(ctx, pathJoin(fs.fsPath, bucket, object))
+		if err != nil && err != errFileAccessDenied {
+			return oi, err
+		}
+		if fi != nil {
+			// If file found and request was with object ending with "/", consider it
+			// as directory and return object info
+			if hasSuffix(object, slashSeparator) {
+				return fsMeta.ToObjectInfo(bucket, object, fi), nil
+			}
+			return oi, errFileNotFound
 		}
-		return oi, errFileNotFound
```

Comment:
Removing this little is little risky because it will just run the next code if we arrive here, can we always return an error here? like unknown error

Reply:
Actually, previous code was confusing, updated it to a simpler flow.